src/main/java/uk/gov/companieshouse/DissolutionApplication.java

```diff
@@ -3,14 +3,20 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.xml.XmlMapper;
 import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Primary;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 @SpringBootApplication
 public class DissolutionApplication {
 
+    @Value("${logging.namespace}")
+    private String loggerNamespace;
```

Comment:
Shouldn't we add this to the `LoggingConfig`? Seems inconsistent the config value and bean is done in the main class, I wouldn't expect to see this here

Reply:
The problem here is that:
* LoggingConfig.java creates binds the LoggingInterceptor
* The LoggingInterceptor.java requires an injectable Logger bean to be created beforehand
* If this Logger Bean creation was moved to the LoggingConfig, then the LoggingConfig would rely on the LoggingInterceptor, which relies on the Bean created in the Logging Config. Circular dependency.
* If the LoggingInterceptor is done in DissolutionApplication instead, unit tests start to fail due to similar issues we seen in the KafkaConfig story.
* I could create separate LoggerConfig and LoggingInterceptorConfig files, but I thought this was a bit overkill.


What do you think?

Reply:
That makes sense. We have other beans in the main class as well which is ok.

Though could we move the `loggerNamespace` value into `LoggingConfig`, and create a getter/setter for it?
And then in the main class, inject it:
```
    private final LoggingConfig loggingConfig;

    @Autowired
    public DissolutionApplication(LoggingConfig loggingConfig) {
        this.loggingConfig = loggingConfig;
    }
```

And then for the bean do
```
@Bean
    public Logger logger() {
        return LoggerFactory.getLogger(loggingConfig.getLoggerNamespace());
    }
```

Means the config stays in the config classes.

Reply:
Tried it but doesn't work. Spring boot complains when running controller unit tests. I'll created 2 config classes to keep the DissolutionApplication file as light as possible

Reply:
ok, separate files also doesn't seem to work. Seems like this is the only combination that works.